src/lib/RelayOrderLib.sol

```diff
@@ -73,37 +102,17 @@ library RelayOrderLib {
 
     /// @notice hash the given order
     /// @param order the order to hash
-    /// @dev We do not hash the entire Input struct as only some of the input information is required in the witness (recipients, and startAmounts). The token and maxAmount are already hashed in the TokenPermissions struct of the permit.
+    /// @dev since token permissions contains the input token's address and amount, we only need to hash the recipient here
```

Comment:
nit: let's add a comment that notes this for both the Input and the FeeEscalator. Ie That we only hash details not included already in the TokenPermissions struct